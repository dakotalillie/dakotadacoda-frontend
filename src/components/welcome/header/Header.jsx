import React from 'react';
import {
  Collapse,
  Container,
  Nav,
  Navbar,
  NavbarBrand,
  NavbarToggler,
  NavItem,
  NavLink
} from 'reactstrap';
import PropTypes from 'prop-types';
import Logo from './logo/Logo';
import './header.css';

class Header extends React.Component {

  static propTypes = {
    handleScroll: PropTypes.func.isRequired
  };

  state = {
    isOpen: false
  };

  toggle = () => {
    this.setState({ isOpen: !this.state.isOpen });
  };

  render() {
    return (
      <div className="header">
        <Container>
          <Navbar color="faded" dark expand="md">
            <NavbarBrand onClick={() => this.props.handleScroll('jumbo')}>
              <Logo />
            </NavbarBrand>
            <NavbarToggler onClick={this.toggle} />
            <Collapse isOpen={this.state.isOpen} navbar>
              <Nav className="ml-auto" navbar>
                <NavItem>
                  <NavLink onClick={() => this.props.handleScroll('jumbo')}>
                    Home
                  </NavLink>
                </NavItem>
                <NavItem>
                  <NavLink onClick={() => this.props.handleScroll('about')}>
                    About
                  </NavLink>
                </NavItem>
                <NavItem>
                  <NavLink onClick={() => this.props.handleScroll('portfolio')}>
                    Projects
                  </NavLink>
                </NavItem>
                <NavItem>
                  <NavLink onClick={() => this.props.handleScroll('resume')}>
                    Skills
                  </NavLink>
                </NavItem>
                <NavItem>
                  <NavLink onClick={() => this.props.handleScroll('blog')}>
                    Blog
                  </NavLink>
                </NavItem>
                <NavItem>
                  <NavLink onClick={() => this.props.handleScroll('contact')}>
                    Contact
                  </NavLink>
                </NavItem>
              </Nav>
            </Collapse>
          </Navbar>
        </Container>
      </div>
    );
  }
}

export default Header;
