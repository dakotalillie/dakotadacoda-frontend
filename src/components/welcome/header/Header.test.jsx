import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Header from './Header';

describe('Header', () => {

  let wrapper, handleScroll;

  beforeEach(() => {
    handleScroll = jest.fn();
    wrapper = shallow(<Header handleScroll={handleScroll} />);
  });

  it('renders successfully', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('clicking on NavbarBrand calls handleScroll with `home` argument', () => {
    wrapper.find('NavbarBrand').simulate('click');
    expect(handleScroll).toHaveBeenLastCalledWith('home');
  });

  it('clicking on NavLink calls handleScroll function from props', () => {
    wrapper.find('#home_nav_link').simulate('click');
    expect(handleScroll).toHaveBeenLastCalledWith('home');
    wrapper.find('#about_nav_link').simulate('click');
    expect(handleScroll).toHaveBeenLastCalledWith('about');
    wrapper.find('#projects_nav_link').simulate('click');
    expect(handleScroll).toHaveBeenLastCalledWith('projects');
    wrapper.find('#skills_nav_link').simulate('click');
    expect(handleScroll).toHaveBeenLastCalledWith('skills');
    wrapper.find('#blog_nav_link').simulate('click');
    expect(handleScroll).toHaveBeenLastCalledWith('blog');
    wrapper.find('#contact_nav_link').simulate('click');
    expect(handleScroll).toHaveBeenLastCalledWith('contact');
  });
});