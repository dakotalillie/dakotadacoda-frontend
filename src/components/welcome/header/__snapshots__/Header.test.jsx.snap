// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders successfully 1`] = `
<div
  className="header"
>
  <Container
    tag="div"
  >
    <Navbar
      color="faded"
      dark={true}
      expand="md"
      tag="nav"
    >
      <NavbarBrand
        onClick={[Function]}
        tag="a"
      >
        <Logo />
      </NavbarBrand>
      <NavbarToggler
        onClick={[Function]}
        tag="button"
        type="button"
      />
      <Collapse
        appear={false}
        enter={true}
        exit={true}
        in={false}
        isOpen={false}
        mountOnEnter={false}
        navbar={true}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        tag="div"
        timeout={350}
        unmountOnExit={false}
      >
        <Nav
          className="ml-auto"
          navbar={true}
          tag="ul"
          vertical={false}
        >
          <NavItem
            key="Home"
            tag="li"
          >
            <NavLink
              id="home_nav_link"
              onClick={[Function]}
              tag="a"
            >
              Home
            </NavLink>
          </NavItem>
          <NavItem
            key="About"
            tag="li"
          >
            <NavLink
              id="about_nav_link"
              onClick={[Function]}
              tag="a"
            >
              About
            </NavLink>
          </NavItem>
          <NavItem
            key="Projects"
            tag="li"
          >
            <NavLink
              id="projects_nav_link"
              onClick={[Function]}
              tag="a"
            >
              Projects
            </NavLink>
          </NavItem>
          <NavItem
            key="Skills"
            tag="li"
          >
            <NavLink
              id="skills_nav_link"
              onClick={[Function]}
              tag="a"
            >
              Skills
            </NavLink>
          </NavItem>
          <NavItem
            key="Blog"
            tag="li"
          >
            <NavLink
              id="blog_nav_link"
              onClick={[Function]}
              tag="a"
            >
              Blog
            </NavLink>
          </NavItem>
          <NavItem
            key="Contact"
            tag="li"
          >
            <NavLink
              id="contact_nav_link"
              onClick={[Function]}
              tag="a"
            >
              Contact
            </NavLink>
          </NavItem>
        </Nav>
      </Collapse>
    </Navbar>
  </Container>
</div>
`;
